test/e2e/suite_test.go

```diff
@@ -941,6 +949,48 @@ func createMultiDatacenterCluster(t *testing.T, ctx context.Context, namespace s
 	assert.NoError(t, err, "timed out waiting for nodetool status check against "+pod)
 }
 
+// createMultiDatacenterDseCluster creates a K8ssandraCluster with three CassandraDatacenters,
+// one running locally and the other two running in a remote cluster.
+// Each CassandraDatacenter is configured with a different DSE workload.
+func createMultiDatacenterDseCluster(t *testing.T, ctx context.Context, namespace string, f *framework.E2eFramework) {
+	require := require.New(t)
+
+	t.Log("check that the K8ssandraCluster was created")
+	k8ssandra := &api.K8ssandraCluster{}
+	kcKey := client.ObjectKey{Namespace: namespace, Name: "test"}
+	err := f.Client.Get(ctx, kcKey, k8ssandra)
+	require.NoError(err, "failed to get K8ssandraCluster in namespace %s", namespace)
+
+	dc2Key := framework.ClusterKey{K8sContext: f.DataPlaneContexts[0], NamespacedName: types.NamespacedName{Namespace: namespace, Name: "dc2"}}
+	checkDatacenterReady(t, ctx, dc2Key, f)
+	assertCassandraDatacenterK8cStatusReady(ctx, t, f, kcKey, dc2Key.Name)
+
+	dc3Key := framework.ClusterKey{K8sContext: f.DataPlaneContexts[0], NamespacedName: types.NamespacedName{Namespace: namespace, Name: "dc3"}}
+	checkDatacenterReady(t, ctx, dc3Key, f)
+	assertCassandraDatacenterK8cStatusReady(ctx, t, f, kcKey, dc2Key.Name, dc3Key.Name)
+
+	dc1Key := framework.ClusterKey{K8sContext: f.DataPlaneContexts[0], NamespacedName: types.NamespacedName{Namespace: namespace, Name: "dc1"}}
+	checkDatacenterReady(t, ctx, dc1Key, f)
+	assertCassandraDatacenterK8cStatusReady(ctx, t, f, kcKey, dc1Key.Name, dc2Key.Name, dc3Key.Name)
+
+	t.Log("retrieve database credentials")
+	username, password, err := f.RetrieveDatabaseCredentials(ctx, f.DataPlaneContexts[0], namespace, k8ssandra.SanitizedName())
+	require.NoError(err, "failed to retrieve database credentials")
+
+	t.Log("check that nodes in dc1 see nodes in dc2 and dc3")
+	pod := DcPrefix(t, f, dc1Key) + "-rack1-sts-0"
```

Comment:
The rack name is wrong since you haven't declared any racks in your spec.

Reply:
👌